app/controllers/pieces_controller.rb

```diff
@@ -35,4 +43,16 @@ def valid_move?
       redirect_to game_path(@piece.game), alert: "This is not a valid move!"
     end
   end
+
+  def can_castle? 
+    piece = Piece.find(params[:piece_id])
+    rook = Piece.find(params[:rook_id])
+    if !piece.can_castle?(rook)
+      redirect_to game_path(piece.game), alert: "You can not Castle at this time!"
```

Comment:
Is it necessary to have this check if we are only displaying the button when castling is possible?

Reply:
I believe so because the link can still be typed in manually.